data/src/main/java/com/teamagam/gimelgimel/data/location/LocationFetcher.java

```diff
@@ -202,18 +208,60 @@ private void validateProviderRegistered() {
 
     private void handleNewLocation(Location location) {
         if (isSufficientQuality(location)) {
-            LocationSample locationSample = convertToLocationSample(location);
-
-            notifyNewLocation(locationSample);
+            notifyNewLocation(location);
         } else {
             notifyNoConnection();
         }
+
+        adjustUpdateFrequency(mOldLocation, location);
+        mOldLocation = location;
```

Comment:
Shouldn't the old-location be updated only if it's a sufficient quality location?